vpr/src/draw/manual_moves.cpp

```diff
@@ -0,0 +1,319 @@
+    int block_id = -1;
+    int x_location = -1;
+    int y_location = -1;
+    int subtile_location = -1;
+    bool valid_input = true;
+
+    t_draw_state* draw_state = get_draw_state_vars();
+
+    //Loading the context/data structures needed.
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+
+    //Getting entry values
+    GtkWidget* block_entry = gtk_grid_get_child_at((GtkGrid*)grid, 0, 1);
+    std::string block_id_string = gtk_entry_get_text((GtkEntry*)block_entry);
+
+    if (string_is_a_number(block_id_string)) { //for block ID
+        block_id = std::atoi(block_id_string.c_str());
+    } else { //for block name
+        block_id = size_t(cluster_ctx.clb_nlist.find_block(gtk_entry_get_text((GtkEntry*)block_entry)));
+    }
+
+    GtkWidget* x_position_entry = gtk_grid_get_child_at((GtkGrid*)grid, 2, 1);
+    GtkWidget* y_position_entry = gtk_grid_get_child_at((GtkGrid*)grid, 2, 2);
+    GtkWidget* subtile_position_entry = gtk_grid_get_child_at((GtkGrid*)grid, 2, 3);
+
+    x_location = std::atoi(gtk_entry_get_text((GtkEntry*)x_position_entry));
+    y_location = std::atoi(gtk_entry_get_text((GtkEntry*)y_position_entry));
+    subtile_location = std::atoi(gtk_entry_get_text((GtkEntry*)subtile_position_entry));
+
+    if (std::string(gtk_entry_get_text((GtkEntry*)block_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)x_position_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)y_position_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)subtile_position_entry)).empty()) {
+        invalid_breakpoint_entry_window("Not all fields are complete");
+        valid_input = false;
+    }
+
+    t_pl_loc to = t_pl_loc(x_location, y_location, subtile_location);
+    valid_input = checking_legality_conditions(ClusterBlockId(block_id), to);
+
+    if (valid_input) {
+        draw_state->manual_moves_global.manual_move_info.valid_input = true;
+        draw_state->manual_moves_global.manual_move_info.blockID = block_id;
+        draw_state->manual_moves_global.manual_move_info.x_pos = x_location;
+        draw_state->manual_moves_global.manual_move_info.y_pos = y_location;
+        draw_state->manual_moves_global.manual_move_info.subtile = subtile_location;
+        draw_state->manual_moves_global.manual_move_info.to_location = to;
+
+        //Highlighting the block
+        deselect_all();
+        ClusterBlockId clb_index = ClusterBlockId(draw_state->manual_moves_global.manual_move_info.blockID);
+        draw_highlight_blocks_color(cluster_ctx.clb_nlist.block_type(clb_index), clb_index);
+        application.refresh_drawing();
+
+        //Continues to move costs window.
+        GtkWidget* proceed = find_button("ProceedButton");
+        ezgl::press_proceed(proceed, &application);
+
+    } else {
+        draw_state->manual_moves_global.manual_move_info.valid_input = false;
+    }
+}
+
+bool checking_legality_conditions(ClusterBlockId block_id, t_pl_loc to) {
```

Comment:
To follow the function naming ... edit it to check_legality_conditions or is_manual_move_legal